tests/unit/FirebaseArray.spec.js

```diff
@@ -122,29 +128,42 @@ describe('$FirebaseArray', function () {
     });
 
     it('should observe $priority and $value meta keys if present', function() {
+      var spy = jasmine.createSpy('$add').and.callFake(function(ref) {
+        expect(ref.priority).toBe(99);
+        expect(ref.getData()).toBe('foo');
+      });
       var arr = stubArray();
-      arr.$add({$value: 'foo', $priority: 99});
-      expect(arr.$$$fb.$push).toHaveBeenCalledWith(jasmine.objectContaining({'.priority': 99, '.value': 'foo'}));
+      arr.$add({$value: 'foo', $priority: 99}).then(spy);
+      flushAll(arr.$ref());
+      expect(spy).toHaveBeenCalled();
+    });
+
+    it('should work on a query', function() {
+      var ref = stubRef();
+      var query = ref.limit(2);
```

Comment:
Look at you using deprecated methods. Should be `limitToLast()`.

On that note, it would also be nice to add methods which add more than two items and then creates `$FirebaseArrray` instances which do `limitToFirst(2)` and `limitToLast(2)` and ensure the correct elements are synced. But we do have an issue tracking these tests so maybe we can just add them later.